app/controllers/products_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ProductsController < ApplicationController
```

Comment:
Products Controller looks good :+1: but I think we may need to differentiate at least two products controllers with the namespace in the routes file because the users and visitors should not be able to access actions like create and edit. We can discuss it later but thanks and good job with the controller outline!